test/shape_inference.jl

```diff
@@ -194,6 +194,21 @@ end
     end
 end
 
+
+@testset "dropout" begin
+    for var in (m,n,k)
+        @test get_shape(nn.dropout(var, 0.5)) == get_shape(var)
+    end
+
+
+    m_by_i = nn.dropout(m, 0.5*i) # a scalar keep_prob
+    @test get_shape(m_by_i) == get_shape(m)
+
+    m_by_n = nn.dropout(m, 0.5*n) # a fully unknown keepprob
+    @test get_shape(m_by_n).rank_unknown == true
```

Comment:
I think we should take it out, or at least comment it out.  That's based on the theory that the point of tests is to ensure that new patches don't interfere with desired and documented behavior; their point isn't just to catalog the way things happen to work now.